startup_script/run.sh

```diff
@@ -32,9 +32,15 @@ CUSTOM_SOURCES_PATH=$(/usr/share/google/get_metadata_value attributes/custom-sou
 # get time to wait for stdout to flush
 SHUTDOWN_TIMER_IN_SEC=$(/usr/share/google/get_metadata_value attributes/shutdown-timer-in-sec)
 
-USER_DATAPROC_COMPONENTS=$( /usr/share/google/get_metadata_value attributes/optional-components | tr '[:upper:]' '[:lower:]' | tr '.' ',' || echo "")
-BDUTIL_DIR="/usr/local/share/google/dataproc/bdutil"
-DATAPROC_VERSION=$(/usr/share/google/get_metadata_value attributes/dataproc-version | cut -c1-3 | tr '-' '.' || echo "")
+USER_DATAPROC_COMPONENTS=$( /usr/share/google/get_metadata_value attributes/optional-components | tr '[:upper:]' '[:lower:]' | tr '.' ' ' || echo "")
+DATAPROC_IMAGE_VERSION=$(/usr/share/google/get_metadata_value attributes/dataproc_dataproc_version | cut -c1-3 | tr '-' '.' || echo "")
+DATAPROC_IMAGE_TYPE=$(/usr/share/google/get_metadata_value attributes/dataproc_image_type || echo "standard")
+export REGION=$(/usr/share/google/get_metadata_value attributes/dataproc-region)
+[[ -n "${DATAPROC_IMAGE_TYPE}" ]] # Sanity validation
+export DATAPROC_IMAGE_TYPE
+[[ "${DATAPROC_IMAGE_VERSION}" =~ ^[0-9]+\.[0-9]+$ ]] # Sanity validation
```

Comment:
this should probably support 2.3-debian12 or 2.3.5-debian12 or 2.3.5-rocky9 or 2.3.5-ubuntu22

Reply:
Ack

Reply:
Upon re-checking, our build scripts expect only "<Major>.<Minor>" format for this env.

For custom image creation, customer can pass in the subminor version as well, which is used to fetch the base image. But this script makes use of only "<major>.<minor>" part of the version.